src/auth.js

```diff
@@ -246,15 +247,40 @@ class Auth {
    * @return {Promise<undefined>}
    */
   async createHomeDirectory(profile) {
-    try {
-      const homeDir = await this
-        .core
-        .make('osjs/vfs')
-        .realpath('home:/', profile);
+    const vfs = this.core.make('osjs/vfs');
+    const template = this.core.config('vfs.home.template', []);
 
-      await fs.ensureDir(homeDir);
-    } catch (e) {
-      console.warn('Failed trying to make home directory for', profile.username);
+    if (typeof template === 'string') {
+      // If the template is a string, it is a path to a directory
+      // that should be copied to the user's home directory
+      const root = await vfs.realpath('home:/', profile);
+
+      fs.copySync(template, root, {overwrite: false});
+    } else if (Array.isArray(template)) {
+      await this.handleHomeDirectoryArray(template, vfs, profile);
+    }
+  }
+
+  /**
+   * If the template is an array, it is a list of files that should be copied
+   * to the user's home directory.
+   * @param {Object[]} template Array of objects with a specified path,
+   * optionally with specified content but defaulting to `[]` if not
+   * @param {VFSServiceProvider} vfs An instance of the virtual file system
+   * @param {AuthUserProfile} profile User profile
+   */
+  async handleHomeDirectoryArray(template, vfs, profile) {
```

Comment:
Rename this to `createHomeDirectoryFromArray`. The `handle` prefix makes things a little big ambiguous in some cases :)